validator/utils/synthetic/synthetic_utils.py

```diff
@@ -109,22 +97,37 @@ async def _get_random_picsum_image(x_dim: int, y_dim: int) -> str:
     return img_b64
 
 
-async def get_random_image_b64(cache: diskcache.Cache) -> str:
-    for key in cache.iterkeys():
-        image_b64: str | None = cache.get(key, None)  # type: ignore
-        if image_b64 is None:
-            cache.delete(key)
-            continue
 
-        if random.random() < 0.01:
-            cache.delete(key)
-        return image_b64
 
+async def get_random_image_b64(redis_db: Redis) -> str:
+    cache_key = scst.IMAGE_CACHE_KEY
+    cache_size = scst.IMAGE_CACHE_SIZE
+
+    logger.debug(f"Starting get_random_image_b64. Cache key: {cache_key}, Cache size: {cache_size}")
+
+    if random.random() < 0.01:
+        if await redis_db.scard(cache_key) > 0:
+            random_key = await redis_db.srandmember(cache_key)
+            await redis_db.srem(cache_key, random_key)
+            await redis_db.delete(random_key)
+    elif await redis_db.scard(cache_key) > 0:
+        random_key = await redis_db.srandmember(cache_key)
+        return await redis_db.get(random_key)
+
+    # If cache is empty or we decided to get a new image
     random_picsum_image = await _get_random_picsum_image(1024, 1024)
-    cache.add(key=str(uuid.uuid4()), value=random_picsum_image)
+    if await redis_db.scard(cache_key) >= cache_size:
```

Comment:
I think it's slightly confusing because of the use of 'scard' etc. 'scard' is a bad name from redis to me because it's not clear whats happening. Overall this func is a little hard to read